zvariant_utils/src/case.rs

```diff
@@ -26,16 +26,60 @@ pub fn pascal_or_camel_case(s: &str, is_pascal_case: bool) -> String {
     result
 }
 
-/// Convert to snake case, assuming pascal case.
+/// Convert to snake case.
 ///
 /// If `s` is already in snake case, should yield the same result.
 pub fn snake_case(s: &str) -> String {
     let mut snake = String::new();
-    for ch in s.chars() {
+    for ch in pascal_or_camel_case(s, true).chars() {
         if ch.is_ascii_uppercase() && !snake.is_empty() {
             snake.push('_');
         }
         snake.push(ch.to_ascii_lowercase());
     }
     snake
 }
+
+/// Convert to kebab case.
+///
+/// If `s` is already in kebab case, should yield the same result.
+pub fn kebab_case(s: &str) -> String {
+    snake_case(s).replace('_', "-")
```

Comment:
This commit ends up parsing and modifying the string 3 times. I'd rather not do that, even if it means a bit of duplication. This is macro code so binary size isn't a concern at least.

Reply:
My assumtion was that this code it is not executed on the hot path. So I didn't pay attention to makeing it fast but rather to make it simple.

Your comment that this is macro code seems to confirm this.

What is your concern with modifying the string multiple times?

Reply:
> My assumtion was that this code it is not executed on the hot path. So I didn't pay attention to makeing it fast but rather to make it simple.

Yeah, it wouldn't be an issue if there was some big benefit from it but I don't see any. I don't think it's simpler with the current approach. It's not more complicated/less-readable with single pass.

Reply:
@zeenix The function is now more explicit. Let me know if this better fits.